### PR TITLE
compat with ModernKeybinding

### PR DESCRIPTION
**Pull request by Caedis**
Integrate the changes from https://github.com/MikhailTapio/ModernControlling/tree/Forge/1.7.10

image

### CODE REVIEW

**Comment by Caedis** [comment]
To be forked/merged in as well
https://github.com/Nova-Committee/ModernKeyBinding/tree/Forge/1.7.10

**Comment by Dream-Master** [comment]
@Caedis what I need to do? Only add the mod? Add the mod as dep to this code change?

**Comment by Caedis** [comment]
Client only mod

**Comment by Dream-Master** [comment]
So just add this together with the mod ?

**Comment by Caedis** [comment]
I am probably going to merge ModernKeybindings into this repo in another PR.

**Comment by Dream-Master** [comment]
Ok. I remember you can set in the run script that it is a client only mod to pass/avoid server test

**Comment by Caedis** [comment]
This is a different issue. Having issues running the mod in java 8

File: src/main/java/com/blamejared/controlling/client/gui/GuiNewControls.java [hunk 1]
```diff
@@ -216,9 +220,16 @@ public void drawScreen(int mouseX, int mouseY, float partialTicks) {
         boolean flag = false;
 
         for (KeyBinding keybinding : this.options.keyBindings) {
-            if (keybinding.getKeyCode() != keybinding.getKeyCodeDefault()) {
-                flag = true;
-                break;
+            if (Controlling.isModernKeybindingInstalled && keybinding instanceof IKeyBinding) {
```
Comment (OneEyeMaker) [inline src/main/java/com/blamejared/controlling/client/gui/GuiNewControls.java]: This is the only place where the `instanceof' check is present. I think such a check should be present everywhere (just for safety). Also, please use modern Java cast syntax if possible.

Comment (Caedis) [reply src/main/java/com/blamejared/controlling/client/gui/GuiNewControls.java]: Should be good now

File: src/main/java/com/blamejared/controlling/client/gui/GuiNewKeyBindingList.java [hunk 2]
```diff
@@ -230,8 +243,9 @@ public void drawEntry(int slotIndex, int x, int y, int listWidth, int slotHeight
                 prefix = EnumChatFormatting.YELLOW + "> " + EnumChatFormatting.RESET + EnumChatFormatting.UNDERLINE;
                 suffix = EnumChatFormatting.RESET.toString() + EnumChatFormatting.YELLOW + " <";
             } else if (hasConflict) {
-                prefix = EnumChatFormatting.RED + "[ " + EnumChatFormatting.RESET;
-                suffix = EnumChatFormatting.RED + " ]";
+                EnumChatFormatting clr = modConflict ? EnumChatFormatting.GOLD : EnumChatFormatting.RED;
+                prefix = clr + "[ " + EnumChatFormatting.RESET;
+                suffix = clr + " ]";
             }
```
Comment (OneEyeMaker) [inline src/main/java/com/blamejared/controlling/client/gui/GuiNewKeyBindingList.java:249]: IMO, if a key binding uses the same key, it should be displayed in red regardless of modifiers.
```suggestion
            } else if (hasConflict) {
                prefix = EnumChatFormatting.RED + "[ " + EnumChatFormatting.RESET;
                suffix = EnumChatFormatting.RED + " ]";
            } else if (modConflict) {
                prefix = EnumChatFormatting.GOLD + "[ " + EnumChatFormatting.RESET;
                suffix = EnumChatFormatting.GOLD + " ]";
            }
```

Comment (Caedis) [reply src/main/java/com/blamejared/controlling/client/gui/GuiNewKeyBindingList.java:249]: Its more of a soft conflict. e binds wont interfere with ctrl+e binds

Comment (Caedis) [reply src/main/java/com/blamejared/controlling/client/gui/GuiNewKeyBindingList.java:249]: Also, you have the wrong logic in your suggestion

Comment (Caedis) [reply src/main/java/com/blamejared/controlling/client/gui/GuiNewKeyBindingList.java:249]: hasConflict will always be true if modConflict is true

**Review by OneEyeMaker — APPROVED** [review]

**Pull request merged**